iidm/iidm-reducer/src/main/java/com/powsybl/iidm/reducer/SubNetworkPredicate.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.reducer;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.*;
+
+import java.util.LinkedHashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A network reducer predicate that allow reduction based on a center voltage level and all other voltage level neighbors
+ * within a specified depth.
+ *
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class SubNetworkPredicate implements NetworkPredicate {
+
+    private String voltageLevelId;
+
+    private int maxDepth;
+
+    private IdentifierNetworkPredicate delegate;
+
+    public SubNetworkPredicate(String voltageLevelId, int maxDepth) {
+        this.voltageLevelId = Objects.requireNonNull(voltageLevelId);
+        if (maxDepth < 0) {
+            throw new IllegalArgumentException("Invalid max depth value: " + maxDepth);
+        }
+        this.maxDepth = maxDepth;
+    }
+
+    private void init(Network network) {
+        if (delegate != null) {
+            return;
+        }
+        VoltageLevel vl = network.getVoltageLevel(voltageLevelId);
+        if (vl == null) {
+            throw new PowsyblException("Voltage level '" + voltageLevelId + "' not found");
+        }
+        Set<String> voltageLevelIds = new LinkedHashSet<>();
+        traverse(vl, 0, maxDepth, voltageLevelIds);
+        delegate = new IdentifierNetworkPredicate(voltageLevelIds);
```

Comment:
I think it's not a good design: the delegate is in fact final (you skip this part if the delegate is already initialized), but initialized after the constructor. I propose to make it delegate final, and initialize the SubNetworkPredicate using a VoltageLevel instead of its ID.

If you want to reuse the SubAreaNetwork for different VL or networks, the `init()` method should return the predicate and you should use a local variable.

Reply:
Ok done